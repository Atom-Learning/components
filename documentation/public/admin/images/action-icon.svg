<svg width="72" height="32" viewBox="0 0 72 32" fill="none" xmlns="http://www.w3.org/2000/svg">
<rect width="32" height="32" rx="4" fill="#0F67F5"/>
<path d="M14.2909 21.3799V21.8699H21.0409H21.5409V21.3699V13.2899V12.7899H21.0409H14.2909H13.7909V13.2899V20.8699V21.3699V21.3799H14.2909ZM10.9508 19.4499C10.9087 19.4318 10.8809 19.3908 10.8809 19.3399V9.99988C10.8809 9.93128 10.9317 9.87988 11.0009 9.87988H19.0009C19.0695 9.87988 19.1209 9.93074 19.1209 9.99988C19.1209 10.0637 19.0647 10.1199 19.0009 10.1199H11.6209H11.1209V10.6199V19.3299C11.1209 19.3985 11.07 19.4499 11.0009 19.4499H10.9508ZM21.6709 22.1199H13.6709C13.6023 22.1199 13.5509 22.069 13.5509 21.9999V12.6699C13.5509 12.6013 13.6017 12.5499 13.6709 12.5499H21.6709C21.7395 12.5499 21.7909 12.6007 21.7909 12.6699V21.9999C21.7909 22.0685 21.74 22.1199 21.6709 22.1199Z" fill="#0F67F5" stroke="white"/>
<path d="M55.978 19.321L55.9844 19.3163L56.0244 19.3563L56.4525 19.7844C56.458 19.747 56.4609 19.7087 56.4609 19.6699C56.4609 19.236 56.1047 18.8799 55.6709 18.8799C55.237 18.8799 54.8809 19.236 54.8809 19.6699C54.8809 20.1037 55.237 20.4599 55.6709 20.4599C55.8966 20.4599 56.1013 20.3635 56.2457 20.2099H55.6709C55.5018 20.2099 55.3875 20.129 55.3438 20.0941C55.2932 20.0536 55.2599 20.012 55.2434 19.9899C55.2246 19.9649 55.2109 19.9426 55.2024 19.9281C55.1979 19.9203 55.1941 19.9134 55.191 19.9077L55.1869 19.8998L55.1852 19.8965L55.1844 19.8949L55.184 19.8942L55.1838 19.8938L55.1837 19.8937C55.1837 19.8936 55.1836 19.8935 55.6309 19.6699L55.1836 19.8935L55.6709 19.5423M55.978 19.321C55.9783 19.3212 55.9786 19.3215 55.9789 19.3218C55.9956 19.336 56.0091 19.3496 56.0196 19.3611C56.0311 19.3738 56.039 19.3841 56.0434 19.3899C56.0546 19.4049 56.0621 19.4172 56.0662 19.4242L56.0725 19.4355L56.0753 19.4408L56.0767 19.4435L56.0774 19.4449L56.0777 19.4456L56.0779 19.4459L56.078 19.4461C56.078 19.4462 56.0781 19.4463 55.7909 19.5899C55.7554 19.6076 55.7156 19.6275 55.6709 19.6499V19.6299V19.5423M55.978 19.321L55.6709 19.5423M55.978 19.321C55.9749 19.3184 55.9718 19.3158 55.9685 19.3132C55.9342 19.2858 55.8299 19.2099 55.6709 19.2099V19.5423M55.978 15.321L55.9844 15.3163L56.0244 15.3563L56.4525 15.7844C56.458 15.747 56.4609 15.7087 56.4609 15.6699C56.4609 15.236 56.1047 14.8799 55.6709 14.8799C55.237 14.8799 54.8809 15.236 54.8809 15.6699C54.8809 16.1037 55.237 16.4599 55.6709 16.4599C55.8966 16.4599 56.1013 16.3635 56.2457 16.2099H55.6709C55.5018 16.2099 55.3875 16.129 55.3438 16.0941C55.2932 16.0536 55.2599 16.012 55.2434 15.9899C55.2246 15.9649 55.2109 15.9426 55.2024 15.9281C55.1979 15.9203 55.1941 15.9134 55.191 15.9077L55.1869 15.8998L55.1852 15.8965L55.1844 15.8949L55.184 15.8942L55.1838 15.8938L55.1837 15.8937C55.1837 15.8936 55.1836 15.8935 55.6309 15.6699L55.1836 15.8935L55.6709 15.5423M55.978 15.321C55.9783 15.3212 55.9786 15.3215 55.9789 15.3218C55.9956 15.336 56.0091 15.3496 56.0196 15.3611C56.0311 15.3738 56.039 15.3841 56.0434 15.3899C56.0546 15.4049 56.0621 15.4172 56.0662 15.4242L56.0725 15.4355L56.0753 15.4408L56.0767 15.4435L56.0774 15.4449L56.0777 15.4456L56.0779 15.4459L56.078 15.4461C56.078 15.4462 56.0781 15.4463 55.7909 15.5899C55.7554 15.6076 55.7156 15.6275 55.6709 15.6499V15.6299V15.5423M55.978 15.321L55.6709 15.5423M55.978 15.321C55.9749 15.3184 55.9718 15.3158 55.9685 15.3132C55.9342 15.2858 55.8299 15.2099 55.6709 15.2099V15.5423M55.978 11.321L55.9844 11.3163L56.0244 11.3563L56.4525 11.7844C56.458 11.747 56.4609 11.7087 56.4609 11.6699C56.4609 11.236 56.1047 10.8799 55.6709 10.8799C55.237 10.8799 54.8809 11.236 54.8809 11.6699C54.8809 12.1037 55.237 12.4599 55.6709 12.4599C55.8966 12.4599 56.1013 12.3635 56.2457 12.2099H55.6709C55.5018 12.2099 55.3875 12.129 55.3438 12.0941C55.2932 12.0536 55.2599 12.012 55.2434 11.9899C55.2246 11.9649 55.2109 11.9426 55.2024 11.9281C55.1979 11.9203 55.1941 11.9134 55.191 11.9077L55.1869 11.8998L55.1852 11.8965L55.1844 11.8949L55.184 11.8942L55.1838 11.8938L55.1837 11.8937C55.1837 11.8936 55.1836 11.8935 55.6309 11.6699L55.1836 11.8935L55.6709 11.5423M55.978 11.321C55.9783 11.3212 55.9786 11.3215 55.9789 11.3218C55.9956 11.336 56.0091 11.3496 56.0196 11.3611C56.0311 11.3738 56.039 11.3841 56.0434 11.3899C56.0546 11.4049 56.0621 11.4172 56.0662 11.4242L56.0725 11.4355L56.0753 11.4408L56.0767 11.4435L56.0774 11.4449L56.0777 11.4456L56.0779 11.4459L56.078 11.4461C56.078 11.4462 56.0781 11.4463 55.7909 11.5899C55.7554 11.6076 55.7156 11.6275 55.6709 11.6499V11.6299V11.5423M55.978 11.321L55.6709 11.5423M55.978 11.321C55.9749 11.3184 55.9718 11.3158 55.9685 11.3132C55.9342 11.2858 55.8299 11.2099 55.6709 11.2099V11.5423" fill="#0F67F5" stroke="#0F67F5"/>
<rect x="40.5" y="0.5" width="31" height="31" rx="3.5" stroke="#0F67F5"/>
</svg>
