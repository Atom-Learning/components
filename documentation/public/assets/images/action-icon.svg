<svg width="72" height="32" fill="none" xmlns="http://www.w3.org/2000/svg"><rect width="32" height="32" rx="4" fill="#0F67F5"/><path d="M14.3 21.38v.49h7.24v-9.08H13.8v8.59h.5Zm-3.35-1.93a.12.12 0 0 1-.07-.11V10c0-.07.05-.12.12-.12h8c.07 0 .12.05.12.12 0 .06-.06.12-.12.12h-7.88v9.21c0 .07-.05.12-.12.12h-.05Zm10.72 2.67h-8a.12.12 0 0 1-.12-.12v-9.33c0-.07.05-.12.12-.12h8c.07 0 .12.05.12.12V22c0 .07-.05.12-.12.12Z" fill="#0F67F5" stroke="#fff"/><path d="m55.98 19.32.04.04.43.42a.79.79 0 0 0-.78-.9.8.8 0 0 0-.79.79c0 .43.36.79.8.79.22 0 .42-.1.57-.25h-.58a.52.52 0 0 1-.43-.22.65.65 0 0 1-.05-.08v-.01l.44-.23-.45.22.5-.35m.3-.22a.45.45 0 0 1 .09.1v.02l-.28.15-.12.06v-.11m.3-.22-.3.22m.3-.22a.48.48 0 0 0-.3-.11v.33m.3-4.22h.01l.04.04.43.42a.79.79 0 0 0-.78-.9.8.8 0 0 0-.79.79c0 .43.36.79.8.79.22 0 .42-.1.57-.25h-.58a.52.52 0 0 1-.43-.22.65.65 0 0 1-.05-.08v-.01l.44-.23-.45.22.5-.35m.3-.22a.45.45 0 0 1 .09.1v.02l-.28.15-.12.06v-.11m.3-.22-.3.22m.3-.22a.48.48 0 0 0-.3-.11v.33m.3-4.22h.01l.04.04.43.42a.79.79 0 0 0-.78-.9.8.8 0 0 0-.79.79c0 .43.36.79.8.79.22 0 .42-.1.57-.25h-.58a.52.52 0 0 1-.43-.22.65.65 0 0 1-.05-.08v-.01l.44-.23-.45.22.5-.35m.3-.22a.45.45 0 0 1 .09.1v.02l-.28.15-.12.06v-.11m.3-.22-.3.22m.3-.22a.48.48 0 0 0-.3-.11v.33" fill="#0F67F5" stroke="#0F67F5"/><rect x="40.5" y=".5" width="31" height="31" rx="3.5" stroke="#0F67F5"/></svg>