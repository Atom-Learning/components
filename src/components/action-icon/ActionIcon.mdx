---
title: Action Icon
component: ActionIcon
description: An accessible interactive wrapper around Icon
category: Navigation
---

`ActionIcon` should be used alongside `Icon` when you require a user interaction.

```tsx preview
<>
  <ActionIcon label="Attach a file" appearance="simple">
    <Icon is={Clip} size="sm" />
  </ActionIcon>
  <ActionIcon label="Attach a file" appearance="subtle">
    <Icon is={Clip} size="sm" />
  </ActionIcon>
  <ActionIcon label="Attach a file" appearance="outline">
    <Icon is={Clip} size="sm" />
  </ActionIcon>
  <ActionIcon label="Attach a file" appearance="solid">
    <Icon is={Clip} size="sm" />
  </ActionIcon>
</>
```

It is recommended to pair `ActionIcon` with `Tooltip` to provide a clear visual description of the action the user is about to perform.

```tsx preview
<Tooltip>
  <Tooltip.Trigger>
    <ActionIcon size="lg" label="Add to basket">
      <Icon is={Basket} size="md" />
    </ActionIcon>
  </Tooltip.Trigger>
  <Tooltip.Content>Add to basket</Tooltip.Content>
</Tooltip>
```

## Polymorphism

The `ActionIcon` component supports polymorphism, therefore depending on whether it receives an `onClick`/`href` as a prop, it will produce a `button` or `link` respectively

```tsx preview
<ActionIcon label="Shuffle selection" onClick={() => console.log('clicked')}>
  <Icon is={Shuffle} size="sm" />
</ActionIcon>
```

```tsx preview
<ActionIcon label="Add a new folder" href="/directory/add">
  <Icon is={FolderAdd} size="sm" />
</ActionIcon>
```
