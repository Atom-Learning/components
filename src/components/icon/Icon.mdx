---
title: Icon
component: Icon
description: A standardised way to render icons
category: Content
---

`Icon` requires an SVG from our icon library and adds standardised `size` options. Review available icons on the [icon listing page](/theme/icons)

## Sizes

```tsx live
<Icon is={Danger} size="sm" />
```

```tsx preview
<Icon is={Eye} size="md" />
```

```tsx preview
<Icon is={Brightness} size="lg" />
```
