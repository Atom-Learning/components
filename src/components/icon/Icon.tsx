import * as React from 'react'

import { styled } from '~/stitches'
import { Override } from '~/utilities/types'

const StyledIcon = styled('svg', {
  color: 'currentcolor',
  variants: {
    size: {
      // TODO: update to use size token when added
      sm: { height: 16, width: 16 },
      md: { height: 24, width: 24 },
      lg: { height: 32, width: 32 }
    }
  }
})

type IconProps = Override<
  React.ComponentPropsWithoutRef<typeof StyledIcon>,
  {
    is: React.FC<React.SVGProps<SVGSVGElement>>
    as: never
  }
>

export const Icon: React.FC<IconProps> = ({
  is: SVG,
  size = 'sm',
  as, // prevent as from being included in ...props and overwriting as={SVG} on line 29
  ...props
}) => (
  <StyledIcon
    as={SVG}
    size={size}
    viewBox="0 0 24 24"
    aria-hidden="true"
    {...props}
  />
)
