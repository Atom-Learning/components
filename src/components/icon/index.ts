export { Icon } from './Icon'
export { Check, Home } from './svg'
