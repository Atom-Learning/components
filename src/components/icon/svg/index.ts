export { ReactComponent as Check } from './check.svg'
export { ReactComponent as Eye } from './eye.svg'
export { ReactComponent as EyeOff } from './eye-off.svg'
export { ReactComponent as Home } from './home.svg'
