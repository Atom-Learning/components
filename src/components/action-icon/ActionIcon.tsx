import type * as Stitches from '@stitches/react'
import { darken } from 'color2k'
import invariant from 'invariant'
import * as React from 'react'

import { styled, theme } from '~/stitches'
import { NavigatorActions } from '~/types'
import { Override } from '~/utilities'

import { Icon } from '../icon/Icon'

const getSimpleVariant = (base: string, interact: string, active: string) => ({
  bg: 'transparent',
  color: base,
  '&:not(:disabled):hover, &:not(:disabled):focus': {
    color: interact
  },
  '&:not(:disabled):active': {
    color: active
  },
  '&[disabled]': {
    color: '$tonal400',
    cursor: 'not-allowed'
  }
})
const getSolidVariant = (base: string, interact: string, active: string) => ({
  bg: base,
  color: 'white',
  '&:not(:disabled):hover, &:not(:disabled):focus': {
    bg: interact,
    color: 'white'
  },
  '&:not(:disabled):active': {
    bg: active
  },
  '&[disabled]': {
    bg: '$tonal100',
    color: '$tonal400',
    cursor: 'not-allowed'
  }
})
const getOutlineVariant = (base: string, interact: string, active: string) => ({
  border: '1px solid',
  borderColor: 'currentColor',
  color: base,
  '&:not(:disabled):hover, &:not(:disabled):focus': {
    color: interact
  },
  '&:not(:disabled):active': {
    color: active
  },
  '&[disabled]': {
    borderColor: '$tonal400',
    color: '$tonal400',
    cursor: 'not-allowed'
  }
})

const StyledButton = styled('button', {
  alignItems: 'center',
  appearance: 'none',
  bg: 'white',
  border: 'unset',
  borderRadius: '$0',
  boxSizing: 'border-box',
  cursor: 'pointer',
  display: 'flex',
  justifyContent: 'center',
  p: 'unset',
  transition: 'all 100ms ease-out',
  variants: {
    theme: {
      neutral: {},
      primary: {},
      success: {},
      warning: {},
      danger: {}
    },
    appearance: {
      simple: {},
      outline: {},
      solid: {}
    },
    size: {
      sm: { size: '$2 ' },
      md: { size: '$3' },
      lg: { size: '$4' },
      xl: { size: '$5' }
    },
    isRounded: {
      true: {
        borderRadius: '$round'
      }
    }
  },
  compoundVariants: [
    // Appearance Simple
    {
      theme: 'neutral',
      appearance: 'simple',
      css: getSimpleVariant('$tonal300', '$primaryMid', '$primaryDark')
    },
    {
      theme: 'primary',
      appearance: 'simple',
      css: getSimpleVariant('$primary', '$primaryMid', '$primaryDark')
    },
    {
      theme: 'primaryDark',
      appearance: 'simple',
      css: getSimpleVariant(
        '$primaryDark',
        darken(theme.colors.primaryDark.value, 0.1),
        darken(theme.colors.primaryDark.value, 0.15)
      )
    },
    {
      theme: 'success',
      appearance: 'simple',
      css: getSimpleVariant('$success', '$successMid', '$successDark')
    },
    {
      theme: 'warning',
      appearance: 'simple',
      css: getSimpleVariant('$warning', '$warningMid', '$warningDark')
    },
    {
      theme: 'danger',
      appearance: 'simple',
      css: getSimpleVariant('$danger', '$dangerMid', '$dangerDark')
    },

    // Appearance Solid
    {
      theme: 'primary',
      appearance: 'solid',
      css: getSolidVariant('$primary', '$primaryMid', '$primaryDark')
    },
    {
      theme: 'primaryDark',
      appearance: 'solid',
      css: getSolidVariant(
        '$primaryDark',
        darken(theme.colors.primaryDark.value, 0.1),
        darken(theme.colors.primaryDark.value, 0.15)
      )
    },
    {
      theme: 'success',
      appearance: 'solid',
      css: getSolidVariant('$success', '$successMid', '$successDark')
    },
    {
      theme: 'warning',
      appearance: 'solid',
      css: getSolidVariant('$warning', '$warningMid', '$warningDark')
    },
    {
      theme: 'danger',
      appearance: 'solid',
      css: getSolidVariant('$danger', '$dangerMid', '$dangerDark')
    },

    // Appearance Outline
    {
      theme: 'primary',
      appearance: 'outline',
      css: getOutlineVariant('$primary', '$primaryMid', '$primaryDark')
    },
    {
      theme: 'primaryDark',
      appearance: 'outline',
      css: getOutlineVariant(
        '$primaryDark',
        darken(theme.colors.primaryDark.value, 0.1),
        darken(theme.colors.primaryDark.value, 0.15)
      )
    },
    {
      theme: 'success',
      appearance: 'outline',
      css: getOutlineVariant('$success', '$successMid', '$successDark')
    },
    {
      theme: 'warning',
      appearance: 'outline',
      css: getOutlineVariant('$warning', '$warningMid', '$warningDark')
    },
    {
      theme: 'danger',
      appearance: 'outline',
      css: getOutlineVariant('$danger', '$dangerMid', '$dangerDark')
    }
  ]
})

type ActionIconProps = Override<
  React.ComponentProps<typeof StyledButton>,
  Stitches.VariantProps<typeof StyledButton> & {
    as?: string | React.ReactNode
    children: React.ReactNode
    label: string
  } & NavigatorActions
>

export const ActionIcon = React.forwardRef<HTMLButtonElement, ActionIconProps>(
  (
    {
      children,
      theme = 'primary',
      appearance = 'simple',
      size = 'md',
      label,
      href,
      disabled,
      ...remainingProps
    },
    ref
  ) => {
    const INVALID_CHILDREN_MESSAGE = `A single ${Icon.displayName} component is permitted as a child of ${ActionIcon.displayName}`

    invariant(React.Children.count(children) === 1, INVALID_CHILDREN_MESSAGE)

    const optionalLinkProps = href
      ? ({
          as: 'a',
          href: disabled ? null : href,
          onClick: undefined,
          'aria-disabled': !!disabled
        } as const)
      : ({ type: 'button' } as const)

    return (
      <StyledButton
        {...remainingProps}
        {...optionalLinkProps}
        aria-label={label}
        theme={theme}
        appearance={appearance}
        size={size}
        ref={ref}
        disabled={disabled}
      >
        {React.Children.map(children, (child) => {
          // TS needs this check for any following code to access child.type
          // even with optional chaining
          if (!React.isValidElement(child)) {
            throw new Error(INVALID_CHILDREN_MESSAGE)
          }

          invariant(
            child.type === Icon,
            `Children of type ${child?.type} aren't permitted. Only an ${Icon.displayName} component is allowed in ${ActionIcon.displayName}`
          )

          return React.cloneElement(child, {
            size:
              child.props.size && appearance === 'simple'
                ? child.props.size
                : size,
            isInActionIcon: appearance !== 'simple',
            css: { ...(child.props.css ? child.props.css : {}) }
          })
        })}
      </StyledButton>
    )
  }
)

ActionIcon.displayName = 'ActionIcon'
