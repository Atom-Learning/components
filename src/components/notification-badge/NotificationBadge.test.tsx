import { render, screen, waitFor } from '@testing-library/react'
import { axe } from 'jest-axe'
import * as React from 'react'

import { NotificationBadge } from '.'
import { ActionIcon } from '../action-icon'
import { Icon } from '../icon'
import { Controls } from '@atom-learning/icons'

describe('NotificationIcon component', () => {
  it('renders an accordion', async () => {
    const { container } = render(
      <NotificationBadge value={3}>
        <ActionIcon appearance="outline" label="testing" size="lg" isRounded>
          <Icon is={Controls} />
        </ActionIcon>
      </NotificationBadge>
    )

    expect(container).toMatchSnapshot()
  })

  it('has no programmatically detectable a11y issues', async () => {
    const { container } = render(
      <NotificationBadge value={3}>
        <ActionIcon appearance="outline" label="testing" size="lg" isRounded>
          <Icon is={Controls} />
        </ActionIcon>
      </NotificationBadge>
    )

    expect(
      await waitFor(() => axe(container), { timeout: 5000 })
    ).toHaveNoViolations()
  })

  it('value is rendered in component', async () => {
    render(
      <NotificationBadge value={3}>
        <ActionIcon appearance="outline" label="testing" size="lg" isRounded>
          <Icon is={Controls} />
        </ActionIcon>
      </NotificationBadge>
    )

    const badge = await screen.findByText('3')

    expect(badge).toBeVisible()
  })
})
