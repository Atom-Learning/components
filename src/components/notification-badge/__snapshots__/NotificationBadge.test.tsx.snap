// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationBadge component renders a NotificationBadge 1`] = `
@media  {
  .c-cmpvrW {
    position: relative;
  }

  .c-dhzjXW {
    display: flex;
  }

  .c-DBukb {
    font-family: var(--fonts-body);
    font-size: var(--fontSizes-sm);
    flex-direction: column;
    justify-content: center;
    position: absolute;
    top: 0;
    right: 0;
    transform: translate(50%, -50%);
    border-radius: var(--radii-round);
    padding: var(--space-1);
    height: var(--sizes-2);
    min-width: var(--sizes-2);
    text-align: center;
    color: #fff;
    background: var(--colors-primary);
  }

  .c-fDzwZw {
    align-items: center;
    -webkit-appearance: none;
    appearance: none;
    background: white;
    border: unset;
    border-radius: var(--radii-0);
    box-sizing: border-box;
    cursor: pointer;
    display: flex;
    justify-content: center;
    padding: unset;
    transition: all 100ms ease-out;
  }

  .c-dbrbZt {
    display: inline-block;
    fill: none;
    stroke: currentcolor;
    stroke-linecap: round;
    stroke-linejoin: round;
    vertical-align: middle;
  }
}

@media  {
  .c-fDzwZw-PrXKS-size-lg {
    height: var(--sizes-4);
    width: var(--sizes-4);
  }

  .c-fDzwZw-rloUt-isRounded-true {
    border-radius: var(--radii-round);
  }

  .c-dbrbZt-hemXWm-size-md {
    height: var(--sizes-2);
    width: var(--sizes-2);
    stroke-width: 2;
  }
}

@media  {
  .c-fDzwZw-jpgoGt-cv {
    border: 1px solid;
    border-color: currentColor;
    color: var(--colors-primary);
  }

  .c-fDzwZw-jpgoGt-cv:not(:disabled):hover,
  .c-fDzwZw-jpgoGt-cv:not(:disabled):focus {
    color: var(--colors-primaryMid);
  }

  .c-fDzwZw-jpgoGt-cv:not(:disabled):active {
    color: var(--colors-primaryDark);
  }

  .c-fDzwZw-jpgoGt-cv[disabled] {
    border-color: var(--colors-tonal400);
    color: var(--colors-tonal400);
    cursor: not-allowed;
  }
}

@media  {
  .c-dbrbZt-ihFPSGE-css {
    height: 20px;
    width: 20px;
  }
}

<div>
  <div
    class="c-PJLV c-cmpvrW"
  >
    <div
      class="c-dhzjXW c-DBukb"
      role="status"
    >
      3
    </div>
    <button
      aria-label="testing"
      class="c-fDzwZw c-fDzwZw-PrXKS-size-lg c-fDzwZw-rloUt-isRounded-true c-fDzwZw-jpgoGt-cv"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="c-dbrbZt c-dbrbZt-hemXWm-size-md c-dbrbZt-ihFPSGE-css"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M17 18V6m-5 12V6M7 18V6M5 8h4m1 8h4m1-4h4"
        />
      </svg>
    </button>
  </div>
</div>
`;
